src/libpgmoneta/wal.c

```diff
@@ -201,6 +201,10 @@ pgmoneta_wal(int srv, char** argv)
          goto error;
       }
 
+      // assign xlogpos at the beginning of the streaming to LSN
+      memset(config->servers[srv].current_wal_lsn, 0, MISC_LENGTH);
+      snprintf(config->servers[srv].current_wal_lsn, MISC_LENGTH, "%s", cmd);
```

Comment:
The `cmd` stuff is weird... just use the new function

Reply:
The new function is used to convert xlogptr to xlogpos, and assign the value to `current_wal_lsn`. Here `cmd` is in fact already the xlogpos.